### DISCUSSION TITLE
Add option to set server MTU

### DISCUSSION THREAD

**Discussion by segevfiner** [wg-easy/wg-easy#1052] [Ideas]
**Describe the bug**
On GCP, the default VPC MTU is 1460, the default MTU of WireGuard is 1420 (1500 - 80) which is too large, a needs to be 1380 (1460 - 80), GCP VPC doesn't support IP fragmentation, which results in fragmented packages being dropped.

**To Reproduce**
1. Setup wg-easy in a VPC network in GCP
2. Try to connect to something which will send large packets, e.g. google.

**Expected behavior**
For there to be a setting for the server MTU to fix this.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux 22.04, macOS 14.4.1 and Windows 11 Pro 10.0.22631
 - Browser Chrome
 - Version 124.0.6367.119
 - Also tried using curl.

**Additional context**
Basically we need a `WG_SERVER_MTU` that just generates the MTU directive into the WireGuard server config file, and we are set AFAIK.

**Comment by peterlewis** [comment]
Does `WG_MTU` not scratch the requisite itch?

**Comment by segevfiner** [comment]
It sets the client MTU, but the server MTU is still set based on logic inside wg-quick. And I had a problem with Docker getting it wrong. I ended up configuring Docker to set the correct MTU for its interfaces in `daemon.json` so wg-quick sets the correct MTU on the server.

**Comment by segevfiner** [comment]
For those wondering, I did it like this in `/etc/docker/daemon.json`:
```json
{
  "mtu": 1460,
  "default-network-opts": {
    "bridge": {
      "com.docker.network.driver.mtu": "1460"
    }
  }
}
```

**Comment by satonotdead** [comment]
It's possible to set the server variable trough wg-easy?

**Comment by silajim** [comment]
I just came across this, it needs to be implemented ASAP

**Comment by yumpyy** [comment]
checking on this, any updates on configuring MTU?

**Reply by kaaax0815** [reply]
this is possible in v15

**Reply by Xynonners** [reply]
> this is possible in v15

how?

**Reply by kaaax0815** [reply]
> how?

Admin Panel > Interface > MTU

**Reply by Xynonners** [reply]
> > how?
> 
> Admin Panel > Interface > MTU

yeah sorry, I wasn't seeing it because pulling :latest from ghcr pulled 14 instead of 15. I see it now.

**Comment by just-mironov** [comment]
Why you @segevfiner decided that overhead is 80? For ipv4 it is 60 bytes, [link](https://lists.zx2c4.com/pipermail/wireguard/2017-December/002201.html)

**Reply by kaaax0815** [reply]
He did not state anywhere that he only uses IPv4. The link you provided even mentions the 80-byte headers for IPv6.

**Reply by segevfiner** [reply]
I just took what wireguard default MTU, which accounts for both IPv4 and IPv6 and adjusted for that.

**Comment by just-mironov** [comment]
I also use gcp and docker, and I have a problem that if I set the mtu **wg0** value other than the virtual interface inside docker, then tls handshake does not occur.

```
root@wireguard:# docker exec -it wg-easy ip -4 -c a | grep mtu
wg0: mtu 1380 qdisc noqueue state UNKNOWN group default qlen 1000
eth0@if283: mtu 1460 qdisc noqueue state UP group default  link-netnsid 0
```

```
root@client:~# curl -vk https://mygame.mygame-preprod.com/
*   Trying 10.100.101.101:443...
* Connected to mygame.mygame-preprod.com (10.100.101.101) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
```
it hangs on this moment.

```
root@wireguard:# docker image ls
ghcr.io/wg-easy/wg-easy   14        d8ad20914b33   14 months ago   148MB
```
